src/pages/_app.tsx

```diff
@@ -0,0 +1,26 @@
+import { AppProps } from 'next/dist/next-server/lib/router/router';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { ReactQueryDevtools } from 'react-query/devtools';
+import { Hydrate } from 'react-query/hydration';
+
+const App = ({ Component, pageProps }:AppProps) => {
+  const queryClient = new QueryClient();
+  queryClient.setDefaultOptions({
+    queries: {
+      staleTime: Infinity,
+      refetchOnWindowFocus: false,
+      refetchInterval: false,
+    },
+  });
```

Comment:
```suggestion
  const queryClient = getQueryClient();
```
```ts
export function getQueryClient () {
  const queryClient = new QueryClient();
  queryClient.setDefaultOptions({
    queries: {
      staleTime: Infinity,
      refetchOnWindowFocus: false,
      refetchInterval: false,
    },
  });
  
  return queryClient;
}
```

지금 당장 중요한 부분은 아니지만, 요런 코드들은 따로 모듈을 나누어 주시는 것도 좋을 것 같습니다!

루트 안에서 한번 이런 행위를 하게되면 나중에 라이브러리를 사용할 때마다 이런 설정 코드들이 여기에 쌓이게 될 거에요. 그리고 꼭 이 행위는 컴포넌트 내부에서 수행되어야 할 이유도 없어보입니다!

그렇다고 어떤 설정은 모듈을 분리해서 하고, 어떤 설정은 루트 안에서 하면 전반적인 추상화 레벨 자체가 맞지 않아서 가독성이 떨어지는 코드가 되어버릴 것 같으니, 저는 아예 처음부터 이런 설정 코드는 분리해놓는 편입니다!

Reply:
config 를 따로 분리하자는 의견 저도 동의합니다! 프로젝트 사이즈가 크지 않을거 같아 파일이 너무 많이 분리 되면 오히려 가독성이 떨어지지 않을까 염려 돼서 `App` 에 포함 시켜 놨는데 확장성을 고려하면 분리 되는게 맞겠네요!

https://github.com/Lubycon/DEVTI-front/pull/18/commits/c364a7e597ed66aec4421818360e9b6598c89ce1